src/Lykke.Job.LogReader/PeriodicalHandlers/AzureLogHandler.cs

```diff
@@ -49,24 +50,33 @@ public IReadOnlyList<TableInfo> GetTableInfo()
 
         public override async Task Execute()
         {
-            if (_tables == null)
+            if (_inProgress)
+                return;
+
+            try
             {
-                _tables = new List<TableInfo>();
-#pragma warning disable 4014
-                FindTables();
-#pragma warning restore 4014
-            }
+                _inProgress = true;
```

Comment:
Race condition possible. I would use SemaphoreSlim instead of bool var or at least add volatile modifer to the _inProgress, change the type to int and use Interlocked.Exchange() to update the value.